thunder/tests/test_auto_register_torchops.py

```diff
@@ -0,0 +1,187 @@
+            continue
+        if device == "cpu" and not torch._C.has_lapack and skipCPUIfNoLapack in op_info.decorators:
+            continue
+        funcs = [name2func[op_info.name], name2func.get(f"Tensor.{op_info.name}", None)]
+        for func in funcs:
+            if func is None:
+                continue
+            total += 1
+            # It takes too long, test only the first 5 sample inputs
+            gen = islice(
+                op_info.sample_inputs_func(
+                    op_info, device=torch.device(device), dtype=torch.float32, requires_grad=requires_grad
+                ),
+                5,
+            )
+            for sample in gen:
+                try:
+                    jfun = thunder.jit(func)
+                    out = jfun(sample.input, *sample.args, **sample.kwargs)
+                except Exception as e:
+                    cnt += 1
+                    assert isinstance(e, NotImplementedError)
+                    assert str(e).startswith(f"Exception encountered when doing automatic registration") or str(
+                        e
+                    ).startswith(f"Unsupported type:")
+                    break
+                else:
+                    if requires_grad:
+                        trc = thunder.last_backward_traces(jfun)[-1]
+                        fwd_trc = thunder.last_traces(jfun)[-1]
+                        # skip if it is not differentiable
+                        outs = fwd_trc.output[0]["output"]
+                        outs = outs if isinstance(outs, tuple) else (outs,)
+                        if all(not thunder.core.dtypes.is_inexact_dtype(o.dtype) for o in outs):
+                            continue
+                        vjp_op_name = f"{op_info.name.split('.')[-1]}_vjp"
+                        if op_info.name == "mm":
+                            assert any(bsym.sym.name.endswith(vjp_op_name) for bsym in trc.bound_symbols)
+                        else:
+                            assert any(bsym.sym.name == vjp_op_name for bsym in trc.bound_symbols)
+                    else:
+                        fwd_trc = thunder.last_traces(jfun)[-1]
+                        assert any(
+                            bsym.sym.name.endswith(op_info.name.split(".")[-1]) and not bsym.subsymbols
+                            for bsym in fwd_trc.bound_symbols
+                        )
+            else:
+                suc += 1
+
+
+# Replace manual registration of some operations with automatic registration for network test cases
+skip_ops_nanogpt = [
+    get_opinfo("layer_norm"),
+    get_opinfo("linear"),
+    get_opinfo("gelu"),
+    get_opinfo("scaled_dot_product_attention"),
+]
+skip_ops_alexnet = [
+    get_opinfo("conv2d"),
+    get_opinfo("linear"),
+    get_opinfo("adaptive_avg_pool2d"),
+    get_opinfo("max_pool2d"),
+]
+disable_opinfos = skip_ops_nanogpt + skip_ops_alexnet
+tmp1 = dict(thunder.core.jit_ext._general_jit_lookaside_map)
```

Comment:
I think it would be nicer to rename it as `tmp_lookaside_map` and similar for `tmp2` and `tmp3`.